cmd/cli/README.md

```diff
@@ -4,40 +4,194 @@
 For now it supports the following commands:
+
 - `evaluate` to evaluate feature flags directly in your terminal
 - `lint` to validate a configuration file format.
 
 ## How to install the cli
 
 ### Install using Homebrew (mac and linux)
+
 ```shell
 brew tap thomaspoignant/homebrew-tap
 brew install go-feature-flag-cli
 ```
 
 ### Install using docker
+
 ```shell
 docker pull gofeatureflag/go-feature-flag-cli
 ```
-More information about the usage of the container in the [dockerhub page](https://hub.docker.com/r/gofeatureflag/go-feature-flag-cli).
+
+More information about the usage of the container in
+the [dockerhub page](https://hub.docker.com/r/gofeatureflag/go-feature-flag-cli).
 
 # How to use the command line
 
 **`go-feature-flag-cli`**  is a command line tool.
 
 ## How to evaluate a flag
+
+The evaluate command allows you to evaluate a feature flag or inspect the configuration of your retriever using
+`--check-mode`
+
+```shell
+go-feature-flag-cli evaluate [OPTIONS]
+```
+
+### Key Flags
+
+| Flag           | Shorthand | Description                                                                                                                     | Default |
+|----------------|-----------|---------------------------------------------------------------------------------------------------------------------------------|---------|
+| `--kind`       | k         | Kind of configuration source. Determines where to read your flags from                                                          | file    |
+| `--config`     | c         | Path to the local flag configuration file (⚠️ deprecated, use `--path` instead)                                                 | ""      |
+| `--format`     | f         | Format of your input file (YAML, JSON or TOML)                                                                                  | yaml    |
+| `--flag`       |           | Name of the flag to evaluate                                                                                                    | ""      |
+| `--timeout`    |           | Timeout in seconds to access your configuration file                                                                            | 0       |
+| `--ctx`        |           | Evaluation context as a json string                                                                                             | {}      |
+| `--check-mode` |           | Check only mode - when set, the command will not perform any <br/>evaluation and returns the configuration of spanned retriever | false   |
+
+Supported values for `--kind` are:
+
+- `file`
+- `http`
+- `github`
+- `gitlab`
+- `s3`
+- `googleStorage`
+- `configmap` (kubernetes)
+- `mongodb`
+- `bitbucket`
+- `azureBlobStorage`
+- `postgresql`
+
+**Caution**: We do not support `redis` retriever as for now due
+to: https://github.com/thomaspoignant/go-feature-flag/issues/4023.
```

Comment:
Do you want to add the serializable layer on this PR or do you prefer to open another pull request for this?

Reply:
This is all good no worries, we can address this later.